src/types/infer.ts

```diff
@@ -1390,15 +1300,15 @@ export class InferType {
                 fieldT
             );
 
+            let builtinT: TypeNode | undefined;
+
             // For solidity <0.5.0 contract variables are implicitly castable to address
             if (lt(this.version, "0.5.0")) {
                 builtinT = addressBuiltins.getFieldForVersion(node.memberName, this.version);
             }
 
             if (fieldT) {
-                if (this.isFunctionCallExternal(node, baseT)) {
-                    fieldT = this.changeLocToMemory(fieldT);
-                }
+                fieldT = this.changeLocToMemory(fieldT);
```

Comment:
Why is this no longer in an if?

Reply:
Well, previous logic actually was working that way... Let me explain:

First of all, this if is nested in
```typescript
if (baseT instanceof UserDefinedType && baseT.definition instanceof ContractDefinition) {
```
So, `baseT` is UserDefinedType, that is a contract...Then we call
```typescript
if (fieldT) {
    if (this.isFunctionCallExternal(node, baseT)) {
```
Where `node` is a `MemberAccess` and `baseT` is UDT of contract. The `isFunctionCallExternal()` back at the time had following implementation:
```typescript
private isFunctionCallExternal(callee: ASTNode, calleeT: TypeNode): boolean {
    if (!(callee instanceof MemberAccess)) { // <- This would never be true
        return false;
    }

    const name = callee.memberName;

    if (
        calleeT instanceof BuiltinFunctionType &&
        ["call", "callcode", "staticcall", "delegatecall", "transfer", "send"].includes(name)
    ) {
        return true;
    }

    const baseT = this.typeOf(callee.vExpression);
    // this is the same thing as we checked in first IF, as initial fieldT was computed in the same way

    return baseT instanceof UserDefinedType && baseT.definition instanceof ContractDefinition;
    // this would always be true
}
```
So, this check seemed to be obsolete. Contract archive run also didn't show any difference.